src/AddressRegistry/StreetName/StreetName.cs

```diff
@@ -328,15 +334,21 @@ public void CorrectAddressBoxNumber(AddressPersistentLocalId addressPersistentLo
         public void RemoveAddress(AddressPersistentLocalId addressPersistentLocalId)
         {
             StreetNameAddresses
-                .GetByPersistentLocalId(addressPersistentLocalId)
+                .GetNotRemovedByPersistentLocalId(addressPersistentLocalId)
```

Comment:
This should stay as it was. Removing an already removed address is seen as an idempotent action.